arch/arm/core/cortex_m/vector_table.S

```diff
@@ -16,7 +16,6 @@
  * rest should not be triggered until the kernel is ready to handle them.
  */
 
-#include <toolchain.h>
```

Comment:
As far as I understand, `toolchain.h` is explicitly included in ASM source files because the macros like `_ASM_FILE_PROLOGUE` are defined by `include/toolchain/gcc.h` which is meant to be included using `include/toolchain.h`.

While this will work because `linker/sections.h` currently includes `toolchain.h`, but there is no guarantee that this will always be the case and I don't think removing this inclusion will be the best approach since ASM toolchain macros are directly depend on it.

Reply:
Fair point @stephanosio, thanks for reviewing.
My intention is to limit inclusions as much as we can - we've been fighting with inclusion cycles in ARM code for quite a long time. I'll see if I can remove linker/sections, instead.